.github/workflows/rustdoc.yml

```
@@ -0,0 +1,42 @@
+name: Build & deploy rustdoc
+
+on:
+  workflow_run:
+    workflows:
+      - CI
+    types:
+      - completed
+    branches:
+      - main
+
+env:
+  CARGO_INCREMENTAL: 0
+  CARGO_NET_RETRY: 10
+  RUSTFLAGS: "-D warnings -W unreachable-pub"
+  RUSTUP_MAX_RETRIES: 10
+
+jobs:
+  rustdoc:
+    if: github.repository == 'Xithrius/twitch-tui' && github.event.workflow_run.conclusion == 'success'
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - name: Install Rust toolchain
+      run: rustup update --no-self-update stable
+
+    - name: Build Documentation
+      run: cargo doc --all --no-deps
+
+    - name: Add index redirection
+      run: echo "<meta http-equiv=\"refresh\" content=\"0; url=twt\">" > ./target/doc/index.html
+
+    - name: Deploy Docs
+      uses: peaceiris/actions-gh-pages@v3
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        publish_branch: main
```

Oh dear. After generating the docs locally I realize that the sheer amount of stuff is too much to be on the main branch.

Sorry for bikeshedding, I'll merge this PR after this change. Thanks for contributing!

```suggestion
        publish_branch: gh-pages
```